src/recovery/tally.rs

```diff
@@ -548,54 +543,89 @@ impl FragmentReplayer {
+                    identifier,
+                    &vote_plan.to_id(),
+                    proposals_idx,
+                    &choice
+                );
+                let res = wallet
+                    .issue_vote_cast_cert(
+                        &self.old_block0_hash,
+                        &self.fees,
+                        &vote_plan,
+                        proposals_idx,
+                        &choice,
+                    )
+                    .unwrap();
+                wallet.confirm_transaction();
+                Ok(res)
+            }
+            Fragment::Transaction(ref transaction) => {
+                let transaction_slice = transaction.as_slice();
+                let (_, identifier, _) = deconstruct_account_transaction(&transaction_slice)?;
+                let address = Identifier::from(identifier.clone())
+                    .to_address(chain_addr::Discrimination::Production);
+
+                let address: Address = address.into();
+                if transaction.nb_outputs() != 1 {
+                    // The wallet lib we use does not corrently expose this functionality
+                    return Err(Error::UnsupportedMultipleOutputs);
+                }
 
-            let choice = if let Payload::Public { choice } = vote_cast.payload() {
-                choice
-            } else {
-                return Err(ValidationError::UnsupportedPrivateVotes.into());
-            };
-
-            let address: Address = address.into();
-            let wallet = self
-                .wallets
-                .get_mut(&address)
-                .ok_or_else(|| Error::NonVotingAccount(address.to_string()))?;
-
-            let vote_plan = self
-                .voteplans
-                .get(vote_cast.vote_plan())
-                .ok_or(Error::MissingVoteplanError)?;
-            let proposals_idx = vote_cast.proposal_index();
-
-            // we still use the old block0 hash because the new ledger will still use the old one for
-            // verifications. This makes possible the usage of old VoteTally transactions without the need
-            // to be replayed.
-            debug!(
-                "replaying vote from {}, vote plan: {}, proposal idx: {}, choice: {:?}",
-                identifier,
-                &vote_plan.to_id(),
-                proposals_idx,
-                &choice
-            );
-            let res = wallet
-                .issue_vote_cast_cert(
+                let output = transaction_slice.outputs().iter().next().unwrap();
+                let output_address = self
+                    .wallets
+                    .get(&output.address.into())
+                    .ok_or_else(|| Error::NonVotingAccount(address.to_string()))?
+                    .address();
+                let wallet = self
+                    .wallets
+                    .get_mut(&address)
+                    .ok_or_else(|| Error::NonVotingAccount(address.to_string()))?;
```

Comment:
For that to happen we need to allow the creation of new account wallets as well, which should not actually be too hard. 
Those new accounts should not be allowed to vote though as per the requirements since they are not in block 0

Reply:
> I think it's valid on the blockchain and may have legitimate uses

But is this practical in the context of Catalyst? I mean, yes, it has its use cases, but we won't be able to distribute rewards to such accounts so there is little to no incentive to do so.

Reply:
What if they vote for change, not for personal reward? :smile: 

Yeah, it's more a wish to cover more of the lower hanging fruits here. It's OK to just report any of the unexpected transactions, to deal with them later if somebody submits them.